lib/sheets.ts

```diff
@@ -0,0 +1,175 @@
+                    reject(err);
+                } else {
+                    resolve(`${result.data} cells cleared.`);
+                }
+            });
+        }));
+    }
+
+    public readRow(row: number) {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.get({
+                range: this.name + "!" + row + ":" + row,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, res: any) => {
+                if (err) {
+                    reject('The API returned an error: ' + err);
+                }
+                const rows = res.data.values;
+                if (rows.length) {
+                    resolve(rows);
+                } else {
+                    reject('No data found.');
+                }
+            });
+        }));
+    }
+
+    public readCell(cell: string) {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.get({
+                range: this.name + "!" + cell,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, res: any) => {
+                if (err) {
+                    reject('The API returned an error: ' + err);
+                }
+                const rows = res.data.values;
+                if (rows.length) {
+                    resolve(rows);
+                } else {
+                    reject('No data found.');
+                }
+            });
+        }));
+    }
+
+    public appendRow(row: string[]) {
+        return _await(new Promise((resolve, reject) => {
+            const values = row;
+            const resource = {
+                values,
+            };
+            sheets.spreadsheets.values.append({
+                range: this.name,
+                resource,
+                spreadsheetId: this.spreadsheetID,
+                valueInputOption: 'RAW',
+            }, (err: any, result: any) => {
+                if (err) {
+                    // Handle error.
+                    reject(err);
+                } else {
+                    resolve(`${result.data.updates.updatedCells} cells appended.`);
+                }
+            });
+        }));
+    }
+}
+
+
+// Factory method to create spreadsheet by name, return spreadsheet class
+export function createSheet(name: string) {
+
+    let newID: string;
+
+    const resource = {
+        properties: {
+            title: name
+        },
+    };
+
+    sheets.spreadsheets.create({
+        fields: 'spreadsheetId',
+        resource,
+    }, (err: any, spreadsheet: any) => {
+        if (err) {
+            // Handle error.
+            log(err);
+        } else {
+            log(spreadsheet.data.spreadsheetId)
+            newID = spreadsheet.data.spreadsheetId;
+        }
+    });
+
+    return new Spreadsheet(newID, name);
+}
+
+// Factory method to get spreadsheet from ID, return spreadsheet class
+export function getSheet(id: string, name: string) {
```

Comment:
I think we should use the API to validate that the sheet actually exists in here. Maybe we drop the `name` parameter and just have ID, so that we can get the spreadsheet from the API and then read the name from there, instead of assuming that the ID and Name match.